akka-actor/src/main/scala/akka/actor/Actor.scala

```diff
@@ -442,6 +442,12 @@ trait Actor {
           "You have to use one of the 'actorOf' factory methods to create a new actor. See the documentation.")
     val c = contextStack.head
     ActorCell.contextStack.set(null :: contextStack)
+    // cache function to `unhandled` to avoid allocations in aroundReceive,
+    // can't use a val in Actor due to binary compatibility restriction
+    c match {
+      case cell: ActorCell ⇒ cell.unhandledFun = unhandled
+      case _               ⇒
```

Comment:
I saw the discussion on gitter about why we don't care - that's fine :)
Please add a small comment here why we don't though